agent/consul/controller/queue/queue.go

```diff
@@ -119,7 +123,7 @@ func (q *queue) Get() (item Request, shutdown bool) {
 
 // Add puts the given Request in the queue. If the Request is already in
 // the queue or the queue is stopping, then this is a no-op.
-func (q *queue) Add(item Request) {
+func (q *queue[T]) Add(item T) {
```

Comment:
Something I've been reflecting on is that `queue[Request].Add` has a natural de-dupliation behavior in OSS because two `Request`s with the same `Kind` and `Name` will hash to the same value. 

In contrast, the new controller request type contains a pointer to `*pbresource.ID` so requests for the same resource will hash to different values.

I'm wondering if we should add a `HashKey` method to `T` to fix this?

Reply:
Fixed in https://github.com/hashicorp/consul/pull/17168